src/components/utilities/document-dialog.tsx

```diff
@@ -2,40 +2,43 @@ import * as React from "react";
 import { Button, Dialog } from "@blueprintjs/core";
 
 interface IProps {
-  id: string;
-  isOpen: boolean;
-  onAccept: (id: string, name: string) => void;
+  parentId?: string;
+  onAccept: (content: string, parentId?: string) => void;
   onClose: () => void;
-  content: string;
+  content?: string;
+  title?: string;
+  prompt?: string;
+  placeholder?: string;
+  maxLength?: number;
 }
 
 interface IState {
   content: string;
 }
 
 export default
-class CommentDialog extends React.Component<IProps, IState> {
+class DocumentDialog extends React.Component<IProps, IState> {
```

Comment:
This component name could be more descriptive. In previous applications I've encountered names like `SingleStringDialog` which gave some indication of the purpose of a given dialog. Something analogous might work here.

Reply:
This reads so much better, by the way